### ISSUE TITLE
Map does not load after returning back from another application

### ISSUE THREAD

**Issue by bkowshik** [nitaliano/react-native-mapbox-gl#353]
### Simulation Steps
1. Use the [Android example](https://github.com/mapbox/react-native-mapbox-gl/blob/master/android/example.js)
2. Click on the map attribution icon at the bottom.
3. Select _Mapbox_ which opens in a browser.
4. Select the `back` button to come back to example app.
5. _The map does not load as expected_.
6. **Clicking on the map area now, will get the map back up.**

![back-from-browser-react-native](https://cloud.githubusercontent.com/assets/2899501/15245540/c3300e1a-1924-11e6-9b8e-ca6abb73601b.gif)

_NOTE: I have got this error on an Android Simulator._

**Comment by bsudekum** [comment]
Interesting, can you confirm this on a physical device too?

**Comment by ghost** [comment]
@bsudekum This is due to the lack of the implementation of Android Lifecycle, I've got it to work by implementing an OnResume handler (using MapBoxGL 4.0 though).

**Comment by jayfunk** [comment]
@AlexLourenco Have you made any more progress on resolving this issue? Did you fully upgrade to using MapBoxGL4.0?

**Comment by ghost** [comment]
@jayfunk Yes, the issue is an easy fix. I have heavily modified the core java code to support clustering but you should be able to add this feature by simply doing something like this:

`
public class ReactNativeMapboxGLManager extends ViewGroupManager<MapView> implements LifecycleEventListener

...

```
@Override
public void onHostResume() {
    mapHandler.getMapView().onResume();
}

@Override
public void onHostPause() {
    mapHandler.getMapView().onPause();
}

@Override
public void onHostDestroy() {
    mapHandler.getMapView().onDestroy();
}
```

`

**Comment by dapetcu21** [comment]
This got implemented in #365.
